pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
@php-coder in your commit comment you mentioned "don't inherit ~30 methods while we're using just two". There's no inheritance in Go, just embedding. See https://golang.org/ref/spec#Struct_types.

When we embed a type `T` in a struct type 'S', by not giving it an explicit name, we cause (some of) its fields and methods to be _promoted_.

It might be just your IDE code-completion giving the feeling that "inherit" ~30 methods vs. "less". The struct type will have the same size either way, again, there's no inheritance.

There's one difference though. Before this patch, the exported type `DefaultCleaner` had fields `FileSystem` and `Docker`, plus all exported fields of those two latter types. After the change, we're potentially breaking users by changing the public exported API of the package, since now `DefaultCleaner` has no exported fields whatsoever.

Were you aware of that? That's the whole point of me bothering to comment :smile: 

I suspect all we might care about is `DefaultCleaner.Cleanup`, so maybe there's no need to revert this.
I really didn't enjoy when I came across similar breaking changes in beego that broke our golang-ex...